### PR TITLE
Remove Validating cache

### PR DESCRIPTION
**Pull request by gkousik**
Since we are not using the mtime based validating cache anywhere and
dont have immediate plans of using it due to performance reasons, I'm
removing this code. This would make it a little bit simpler to update
the cache entry of the output files.

Test: Unit tests

### CODE REVIEW

**Comment by ola-rozenfeld** [comment]
Please resolve conflicts. Also: why the command_proto change?

File: go/api/command/BUILD.bazel [hunk 1]
```diff
@@ -12,7 +12,7 @@ proto_library(
 go_proto_library(
     name = "command_go_proto",
     importpath = "github.com/bazelbuild/remote-apis-sdks/go/api/command",
-    proto = ":command_proto",
+    proto = ":cmd_proto",
```
Comment (ola-rozenfeld) [inline go/api/command/BUILD.bazel:15]: Why this change?

Comment (gkousik) [reply go/api/command/BUILD.bazel:15]: This is something that gazelle did automatically. I'll revert it from this change, but perhaps that's a naming convention that gazelle thinks is correct.

**Comment by gkousik** [comment]
Resolved merge conflicts.
